backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java

```diff
@@ -46,18 +55,25 @@ protected void get(HttpServletRequest request, HttpServletResponse response) thr
   @Override
   protected void post(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String ingredients = request.getReader().readLine();
-    String json = Jsoup.connect(BBCGoodFoodRecipeScraper.searchRecipeLink(ingredients, key))
-      .ignoreContentType(true).execute().body();
+    String json =
+        Jsoup.connect(BBCGoodFoodRecipeScraper.searchRecipeLink(ingredients, key))
+            .ignoreContentType(true)
+            .execute()
+            .body();
     JsonObject jsonObject = new JsonParser().parse(json).getAsJsonObject();
     JsonArray items = jsonObject.get("items").getAsJsonArray();
 
+    Pair<List<String>, List<String>> dietsAndAllergies = getUserDietsAndAllergies();
+    List<String> diets = dietsAndAllergies.getLeft();
+    List<String> allergies = dietsAndAllergies.getRight();
```

Comment:
Creating a pair only to unwrap it immediately doesn't look too good. How about this: create 2 new methods `getUserEntity()` and `getPropertyAsList(Entity,String)`. Then here, you'd do:

```
Entity entity = getUserEntity();
List<String> diets = getPropertyAsList(entity, UserConstants.PROPERTY_DIETS);
List<String> allergies = getPropertyAsList(entity, UserConstants.PROPERTY_ALLERGIES);
```

Below, you'd have:
```
private Entity getUserEntity() {
  UserService userService = UserServiceFactory.getUserService();
  String userId = userService.getCurrentUser().getUserId();
  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
  return UserCollector.getUserEntity(userId, datastore);
}
```

And then you could even use a generic list type instead of only Strings:
```
private <T> List<T> getPropertyAsList(Entity entity, String property) {
  List<T> list = (List<T>) entity.getProperty(property);
  if (list == null) {
    return Collections.emptyList();
  }
  return list;
}
```
(I didn't actually compile this to see if the syntax is 100% correct but it should work)

Reply:
I created a DatastoreUtils class that has the suggested methods. I realised that other servlets might also need to use these methods, so I plan to do a follow up CL to refactor servlets.